docs/how-to-work-on-localized-client-webapp.md

```diff
@@ -0,0 +1,270 @@
+```js
+import { useTranslation } from 'react-i18next';
+
+// in the render method:
+const { t } = useTranslation();
+
+// call the "t" function with a key from the JSON file:
+<p>{t('key')}</p>; // more details below
+```
+
+### Class Component
+
+```js
+import { withTranslation } from 'react-i18next';
+
+// withTranslation adds the "t" function to props:
+const { t } = this.props;
+
+// call the "t" function with a key from the JSON file:
+<h1>{t('key')}</h1> // more details below
+
+// export without redux:
+export default withTranslation()(Component);
+
+// or with redux:
+export default connect(...)(withTranslation()(Component));
+```
+
+## Translate Using the "t" Function
+
+### Basic Translation
+
+```js
+// in the component:
+<p>{t('p1')}</p>
+
+// in the JSON file:
+{
+  "p1": "My paragraph"
+}
+
+// output:
+<p>My paragraph</p>
+```
+
+### With Dynamic Data
+
+```js
+// in the component:
+const username = 'moT';
+
+<p>{t('welcome', { username: username })}</p>
+
+// in the JSON file:
+{
+  "welcome": "Welcome {{username}}"
+}
+
+// output:
+<p>Welcome moT</p>
+```
+
+The above example passes an object to the `t` function with a `username` variable. The variable will be used in the JSON value where `{{username}}` is.
+
+## Translate with the `Trans` Component
+
+The general rule is to use the "t" function when you can. But there's a `Trans` component for when that isn't enough, usually when you have elements embedded in the text. You can use the `Trans` component with any type of react component.
+
+### Basic Elements Nested
+
+```js
+// in the component:
+import { Trans } from 'react-i18next'
+
+<p>
+  <Trans>fcc.greeting</Trans>
+</p>
+
+// in the JSON file:
+{
+  "fcc": {
+    "greeting": "Welcome to <strong>freeCodeCamp</strong>"
+  }
+}
+
+// output:
+<p>Welcome to <strong>freeCodeCamp</strong></p>
+```
+
+You can place the key inside the component tags like the above example if the text contains "simple" tags with no attributes. `br`, `strong`, `i`, and `p` are the default, but that list can be expanded in the i18n config.
+
+### Complex Elements Nested
+
+Other times, you will want to have certain text inside another element, an anchor tag is a good example:
+
+```js
+// in the component:
+<p>
+  <Trans i18nKey='check-forum'>
+    <a href='https://forum.freecodecamp.org/'>placeholder</a>
+  </Trans>
+</p>
+
+// in the JSON file:
+{
+  "check-forum": "Check out <0>our forum</0>."
+}
+
+// output:
+<p>Check out <a href='https://forum.freecodecamp.org/'>our forum</a></p>
+```
+
+In the above example, the key is set in the attributes of the `Trans` component. The `<0>` and `</0>` in the JSON represent the first child of the component, in this case, the anchor element. If there were more children, they would just count up from there using the same syntax. You can find the children of a component in the react dev tools by inspecting it. `placeholder` is simply there because the linter was complaining at me about an empty `<a>` element.
```

Comment:
```suggestion
In the above example, the key is set in the attributes of the `Trans` component. The `<0>` and `</0>` in the JSON represent the first child of the component, in this case, the anchor element. If there were more children, they would just count up from there using the same syntax. You can find the children of a component in the react dev tools by inspecting it. `placeholder` is simply there because the linter complains about empty `<a>` elements.
```